src/host/history.cpp

```diff
@@ -199,7 +200,7 @@ std::wstring_view CommandHistory::RetrieveNth(Index index)
         return {};
     }
 
-    LastDisplayed = std::clamp(index, 0, GetNumberOfCommands() - 1);
+    LastDisplayed = (index + GetNumberOfCommands()) % GetNumberOfCommands();

```

Comment:
At a glance this change looks correct to me, but it probably only works if you remove the new if branch above. If you have any questions about this code, please let me know!